atlasdb-api/src/main/java/com/palantir/atlasdb/spi/AtlasDbFactory.java

```diff
@@ -38,6 +38,13 @@
 
     String getType();
 
+    default KeyValueServiceConfig createMergedKeyValueServiceConfig(
```

Comment:
If we really wanted to make this nice, we could parameterize `AtlasDbFactory` by the config class so we wouldn't need to cast.

Reply:
yeah but look how they catch class cast in https://github.com/palantir/atlasdb/blob/develop/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java#L92

Reply:
How is that related to my suggestion?

Reply:
you'd have to move the class cast catching outside of the factory then

Reply:
I'm just talking about the install config here (the type returned by `createMergedKeyValueServiceConfig`.

you're link above is for the runtime config, that can stay as is.

Reply:
This might be more trouble than it's worth. But it would help force callers to use `createMergedKeyValueServiceConfig` instead of accidentally passing in the un-merged install config.

Reply:
`atlasFactory.createRawKeyValueService` doesn't know what type it is so it'll be a raw type....but maybe it'll help implementations understand what's going on